L1Trigger/Phase2L1GT/plugins/L1GTSingleCollectionCut.h

```diff
@@ -145,10 +150,39 @@ namespace l1t {
       desc.add<std::vector<double>>("regionsMinPt", {});
       desc.add<std::vector<double>>("regionsMaxIso", {});
       desc.add<std::vector<unsigned int>>("regionsQual", {});
+      desc.addOptional<double>("minPrimVertDz");
+      desc.addOptional<double>("maxPrimVertDz");
+      desc.addOptional<unsigned int>("primVertex");
     }
 
     const edm::InputTag& tag() const { return tag_; }
 
+  private:
+    bool checkEtaDependentCuts(const P2GTCandidate& obj) const {
+      bool res = true;
+
+      unsigned int index;
+      index = atIndex(obj.hwEta());
+      res &= regionsMinPt_.empty() ? true : obj.hwPT() > regionsMinPt_[index];
+      res &= regionsMaxIso_.empty() ? true
+                                    : obj.hwIso().to_int64() << scales_.isolation_shift() <
+                                          static_cast<int64_t>(regionsMaxIso_[index]) * obj.hwPT().to_int64();
+      res &= regionsQual_.empty() ? true : (obj.hwQual().to_uint() & regionsQual_[index]) == regionsQual_[index];
+      return res;
+    }
+
+    unsigned int atIndex(int objeta) const {
+      // Function that checks at which index the eta of the object is
+      // If object abs(eta) < regionsAbsEtaLowerBounds_[0] the function returns the last index,
+      // Might be undesired behaviour
+      for (unsigned int i = 0; i < regionsAbsEtaLowerBounds_.size() - 1; i++) {
+        if (std::abs(objeta) >= regionsAbsEtaLowerBounds_[i] && std::abs(objeta) < regionsAbsEtaLowerBounds_[i + 1]) {
+          return i;
+        }
+      }
+      return regionsAbsEtaLowerBounds_.size() - 1;
+    }
```

Comment:
I would guess the performance details of this don't matter, but would it be better to binary search this instead of linearly?